src/helpers/__tests__/ci.test.ts

```diff
@@ -45,13 +89,51 @@ describe('getCIMetadata', () => {
       expect(getCIMetadata()).toEqual(expectedMetadata)
     })
   })
+
+  describe.each(CI_PROVIDERS)('Ensure DD env variables override %s env variables', (ciProvider) => {
+    const DD_METADATA = {
+      DD_CI_JOB_NAME: 'DD_CI_JOB_NAME',
+      DD_CI_JOB_URL: 'DD_CI_JOB_URL',
+      DD_CI_PIPELINE_ID: 'DD_CI_PIPELINE_ID',
+      DD_CI_PIPELINE_NAME: 'DD_CI_PIPELINE_NAME',
+      DD_CI_PIPELINE_NUMBER: 'DD_CI_PIPELINE_NUMBER',
+      DD_CI_PIPELINE_URL: 'DD_CI_PIPELINE_URL',
+      DD_CI_PROVIDER_NAME: 'DD_CI_PROVIDER_NAME',
+      DD_CI_STAGE_NAME: 'DD_CI_STAGE_NAME',
+      DD_CI_WORKSPACE_PATH: 'DD_CI_WORKSPACE_PATH',
+      DD_GIT_BRANCH: 'DD_GIT_BRANCH',
+      DD_GIT_COMMIT_AUTHOR_DATE: 'DD_GIT_COMMIT_AUTHOR_DATE',
+      DD_GIT_COMMIT_AUTHOR_EMAIL: 'DD_GIT_COMMIT_AUTHOR_EMAIL',
+      DD_GIT_COMMIT_AUTHOR_NAME: 'DD_GIT_COMMIT_AUTHOR_NAME',
+      DD_GIT_COMMIT_COMMITTER_DATE: 'DD_GIT_COMMIT_COMMITTER_DATE',
+      DD_GIT_COMMIT_COMMITTER_EMAIL: 'DD_GIT_COMMIT_COMMITTER_EMAIL',
+      DD_GIT_COMMIT_COMMITTER_NAME: 'DD_GIT_COMMIT_COMMITTER_NAME',
+      DD_GIT_COMMIT_MESSAGE: 'DD_GIT_COMMIT_MESSAGE',
+      DD_GIT_COMMIT_SHA: 'DD_GIT_COMMIT_SHA',
+      DD_GIT_REPOSITORY_URL: 'DD_GIT_REPOSITORY_URL',
+      DD_GIT_TAG: 'DD_GIT_TAG',
+    }
+
+    const expectedMetadata = ciAppTagsToMetadata(ddMetadataToSpanTags(DD_METADATA))
+    delete expectedMetadata.git.branch
```

Comment:
why delete the branch?

Reply:
`git.branch` is the only edge case and can break depending of tests cases